### ISSUE TITLE
RNN BucketSentenceIter - Must load all data into memory?

### ISSUE THREAD

**Issue by wcollins-ebsco** [apache/incubator-mxnet#8191]
Is there a way to use BucketSentenceIter without having to load all the data into memory? I'm trying to iterate over GB of text.

For bugs or installation issues, please provide the following information.
The more information you provide, the more likely people will be able to help you.

## Environment info
Operating System: docker running ubuntu

Compiler:

Package used (Python/R/Scala/Julia): python

MXNet version: 

Or if installed from source: cloned repo

MXNet commit hash (`git rev-parse HEAD`):

If you are using python package, please provide

Python version and distribution: 2.7

If you are using R package, please provide

R `sessionInfo()`:

## Error Message:
Please paste the full error message, including stack trace.

## Minimum reproducible example
if you are using your own code, please provide a short script that reproduces the error.

## Steps to reproduce
or if you are running standard examples, please provide the commands you have run that lead to the error.

1.
2.
3.

## What have you tried to solve it?

1.
2.
3.

**Comment by ZiyueHuang** [comment]
Currently in BucketSentenceIter there is no way. But you can write your custom bucket data iterator.

**Comment by szha** [comment]
@apache/mxnet-committers: This issue has been inactive for the past 90 days. It has no label and needs triage.

For general "how-to" questions, our [user forum](https://discuss.mxnet.io/) (and [Chinese version](https://discuss.gluon.ai/)) is a good place to get help.

**Comment by vandanavk** [comment]
@wcollins-ebsco were you able to find a way to do this?

**Comment by wcollins-ebsco** [comment]
Thank you for your follow up.
I switched to using scikit-learn and tensorflow due to this and many other problems I ran into while trying to use mxnet.
